packages/utils/src/cache/storages/indexeddb.ts

```diff
@@ -0,0 +1,177 @@
+import { openDB } from 'idb';
+import type { DBSchema, IDBPDatabase } from 'idb';
+import type { CacheMap } from '../types';
+import type { CacheItem } from '../interfaces/CacheItem';
+import type { CacheStorage } from '../interfaces/CacheStorage';
+import { StoragePrefix } from '../constants.js';
+import { getItemKey } from '../helpers.js';
+
+interface IndexedDBDatabase extends DBSchema {
+  [key: StoreName]: {
+    key: string;
+    value: CacheItem;
+  }
+}
+
+/**
+ * Literal type for dynamic store name casting to idb
+ */
+type StoreName = `[${StoragePrefix.DEFAULT}][${string}]`;
+
+/**
+ * Returns Error message when unable to connect indexedDB
+ * Database and Store are same name. Version is 1. So don't try to log a real database config.
```

Comment:
Not sure what this means.

Reply:
I want to say what database, store, and version that the util try to access. Because Indexeddb allows only access latest version. To know current version, `indexeddb..databases()` is native function to know all database version isn't support in firefox.

Reply:
Please revise. I do not get what you are trying to write.

Reply:
I want to log database, store, and version that the util try to access that will be database name equal to store name, version is fit to 1, and it has only 1 store in the database. So I think the errorMessage should log what attempt connection. It don't log current version.

To log current version, I haven't found the way that support all browsers. `window.indexeddb.databases()` is one way to get current version of all databases, but it's not support on Firefox.

Reply:
I would rephrase to something shorter and rewrite the comment. Does something like `window.indexeddb.databases() ? window.indexeddb.databases() : 1` work ?

Reply:
Updated. As we talked, It already said in open. So the comment line is removed.